### ISSUE TITLE
mkbundled'ed executable depends on argv [0] being populated in full, does not work via $PATH look-up.

### ISSUE THREAD

**Issue by HinTak** [mono/mono#10470]
## Steps to Reproduce

1. build a self-contained executable via mkbundle with mono 4.8.x or above

let's call that HelloWorld:

```
mkbundle ... -o  HelloWorld HelloWorld.exe
```

2. put it somewhere in $PATH

i.e. say, put `HelloWorld` into `/usr/bin` .

3. invoke it implicitly via just its basename.

i.e. by just typing `HelloWord` in the terminal.

### Current Behavior

mono cannot find the binary itself to unpack:

```
Usage is: mono [options] program [program-options]

Development:
    --aot[=<options>]      Compiles the assembly to native code
    --debug[=<options>]    Enable debugging support, use --help-debug for details
    --debugger-agent=options Enable the debugger agent
    --profile[=profiler]   Runs in profiling mode with the specified profiler module
    --trace[=EXPR]         Enable tracing, use --help-trace for details
    --jitmap               Output a jit method map to /tmp/perf-PID.map
    --help-devel           Shows more options available to developers

Runtime:
    --config FILE          Loads FILE as the Mono config
    --verbose, -v          Increases the verbosity level
    --help, -h             Show usage information
    --version, -V          Show version information
    --runtime=VERSION      Use the VERSION runtime, instead of autodetecting
    --optimize=OPT         Turns on or off a specific optimization
                           Use --list-opt to get a list of optimizations
    --security[=mode]      Turns on the unsupported security manager (off by default)
                           mode is one of cas, core-clr, verifiable or validil
    --attach=OPTIONS       Pass OPTIONS to the attach agent in the runtime.
                           Currently the only supported option is 'disable'.
    --llvm, --nollvm       Controls whenever the runtime uses LLVM to compile code.
    --gc=[sgen,boehm]      Select SGen or Boehm GC (runs mono or mono-sgen)
```

### Expected Behavior

Mono should be able to unpack the assembly and execute it.

## On which platforms did you notice this

[*] macOS
[*] Linux
[ ] Windows

**Version Used**:

```
Mono JIT compiler version 4.8.0 (Stable 4.8.0.524/9d74414 Mon Apr  3 17:33:49 UTC 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen

```

## Stacktrace

```
Please paste the stack trace here if available.
```

**Comment by jaykrell** [comment]
Mono has code to get its own full path.
Posix does not offer this so there is platform-specific code.

**Comment by HinTak** [comment]
@migueldeicaza

The source of the issue is as in 

https://github.com/mono/mono/commit/66a493cefe3eae0b8a136f6a046f01bb76be33b0#diff-4cc1ebc118f2764dfdcf9ddb892e0854

where `mono/mini/main.c` , `probe_embedded` is introduced. It is called with 
`probe_embedded (argv [0], &argc, &argv)` . And that's pretty much it.

**Comment by HinTak** [comment]
The issue was first noticed in https://github.com/googlefonts/fontbakery/issues/1928 .

**Comment by jaykrell** [comment]
On Linux `readlink(/proc/self/exe)`

**Comment by HinTak** [comment]
The problem is not symbolic links, but calling via basename and $PATH . When an executable is run via shell's $PATH look-up, `argv[0]` is just the basename.

**Comment by jaykrell** [comment]
I understand. `readlink(/proc/self/exe)` provides `ARGV[0]` as a full path.

**Comment by HinTak** [comment]
Argh, sorry, wasn't reading properly. Yes, `readlink(/proc/self/exe)` could work - for linux. Mainly needs it for mac os x though :-).

**Comment by HinTak** [comment]
argh, another sorry. problem already fixed. It landed somewhere between `mono-5.10.1.57` and `mono-5.12.0.226` . The first tag I found with the fix is `mono-5.12.0.226`.

```
commit 947d0bc0ef63c9e9304b760d333b4cb7ed06df94
Author: Ramin Zaghi at Mosaic3DX.com <rzaghi@mosaic3dx.com>
Date:   Thu Jan 18 18:31:20 2018 +0000

    [mini] Make sure mkbundle executables work if they are run from a path on the PATH environment variable. (#6147)
    
    * Make sure mkbundle executables work if they are run from a path on the PATH environment variable.
    
    * Review
    
    * Fix build on Windows

```

I'll see how mono 5.12 behaves...
